store/tikv/gcworker/gc_worker.go

```diff
@@ -495,6 +514,22 @@ func (w *GCWorker) loadGCConcurrencyWithDefault() (int, error) {
 	return jobConcurrency, nil
 }
 
+func (w *GCWorker) loadGCEnableStatus() (bool, error) {
+	str, err := w.loadValueFromSysTable(gcEnableKey)
```

Comment:
Is this and updating safepoint in the same transaction? I'm afraid that you may update `gcEnable` after gc_worker checking it but before gc_worker updating safepoint. So when you disable gc and check the safepoint, gc may be still running and updates the safepoint after you check it.

Reply:
There is no transaction  here. There will only read `gcEnable` from `mysql.tidb`, and if there is no `gcEnable` in `mysql.tidb`, will insert `gcEnable` with default value to `mysql.tidb`, This will only occur at tidb start up or this new tidb version become gc leader. 
Check `gcEnable` is before check `safepoint`.

Reply:
So what if:

* You get the safepoint, suppose it equals `x`
* The gc_leader checks gcEnable, got `true`, continue
* You set gcEnable to `false`, and then use the safepoint `x` to do somethins
* The gc_leader updates safepoint to `y`, and soon it does gc on the whole cluster.

May this happen? Or, do you have other approach to avoid it?

Reply:
Oh,I got you.

Reply:
I have add a transaction for check gcEnable and update safepoint now. PTAL

Reply:
Ok. It seems to be fine. Please also add some comments to explain why we need a transaction there.

Reply:
done.